src/ophyd_async/fastcs/panda/_trigger.py

```diff
@@ -51,6 +70,110 @@ async def stop(self):
         await wait_for_value(self.seq.active, False, timeout=1)
 
 
+@dataclass
+class PosOutScaleOffset:
+    name: str
+    scale: SignalRW[float]
+    offset: SignalRW[float]
+
+    @classmethod
+    def from_inenc(cls, panda: CommonPandaBlocks, number: int) -> PosOutScaleOffset:
+        inenc = panda.inenc[number]  # type: ignore
+        return cls(
+            name=f"INENC{number}.VAL",
+            scale=inenc.val_scale,  # type: ignore
+            offset=inenc.val_offset,  # type: ignore
+        )
+
+
+class ScanSpecSeqTableTriggerLogic(FlyerController[ScanSpecInfo]):
+    def __init__(
+        self, seq: SeqBlock, motor_pos_outs: dict[Motor, PosOutScaleOffset]
+    ) -> None:
+        self.seq = seq
+        self.motor_pos_outs = motor_pos_outs
+
+    async def prepare(self, value: ScanSpecInfo):
+        await self.seq.enable.set(PandaBitMux.ZERO)
+        slice = Path(value.spec.calculate()).consume()
+        slice_duration = error_if_none(slice.duration, "Slice must have duration")
+
+        gaps = np.where(slice.gap)[0]
+        if gaps[0] == 0:
+            gaps = np.delete(gaps, 0)
+        scan_size = len(slice)
+
+        gaps = np.append(gaps, scan_size)
+        fast_axis = slice.axes()[-1]
+        if not self.motor_pos_outs.get(fast_axis):
+            raise RuntimeError("Failed to fetch motor")
+        else:
+            motor_pos_outs = self.motor_pos_outs
+            _, scale, _ = (
+                motor_pos_outs[fast_axis].name,
+                await motor_pos_outs[fast_axis].scale.get_value(),
+                await motor_pos_outs[fast_axis].offset.get_value(),
+            )
+            resolution = scale
+
+        start = 0
+
+        # GPIO goes low
+        rows = SeqTable.row(trigger=SeqTrigger.BITA_0)
+        for gap in gaps:
+            # GPIO goes high
+            rows += SeqTable.row(trigger=SeqTrigger.BITA_1)
+            # Wait for position
+            if (
+                slice.midpoints[fast_axis][gap - 1] * resolution
+                > slice.midpoints[fast_axis][start] * resolution
+            ):
+                trig = SeqTrigger.POSA_GT
+                dir = False if resolution > 0 else True
+
+            else:
+                trig = SeqTrigger.POSA_LT
+                dir = True if resolution > 0 else False
```

Comment:
I think we should use the lower->midpoint to determine the direction, and if they are equal we can't insert a trigger position:
```suggestion
            lower = (slice.lower[fast_axis][start] - offset) / scale
            midpoint = (slice.midpoint[fast_axis][start] - offset) / scale            
            if midpoint > lower:
                trig = SeqTrigger.POSA_GT
            elif midpoint < lower:
                trig = SeqTrigger.POSA_LT
            else:
                trig = None
```